wgsl/index.bs

```diff
@@ -445,60 +445,71 @@ A <dfn>token</dfn> is a contiguous sequence of characters forming one of:
 
-    | `/((-?[0-9]*\.[0-9]+|-?[0-9]+\.[0-9]*)((e|E)(\+|-)?[0-9]+)?)|(-?[0-9]+(e|E)(\+|-)?[0-9]+)/`
+    | `/(0x[0-9a-fA-F]+|0|[1-9][0-9]*)[iu]?/`
 </div>
-<div class='syntax' noexport='true'>
-  <dfn for=syntax>hex_float_literal</dfn> :
 
-    | `/-?0x((([0-9a-fA-F]*\.[0-9a-fA-F]+|[0-9a-fA-F]+\.[0-9a-fA-F]*)((p|P)(\+|-)?[0-9]+)?)|([0-9a-fA-F]+(p|P)(\+|-)?[0-9]+))/`
-</div>
 <div class='syntax' noexport='true'>
-  <dfn for=syntax>int_literal</dfn> :
+  <dfn for=syntax>decimal_float_literal</dfn> :
 
-    | `/-?0x[0-9a-fA-F]+|0|-?[1-9][0-9]*/`
+    | `/((([0-9]*\.[0-9]+|[0-9]+\.[0-9]*)((e|E)(\+|-)?[0-9]+)?)|([0-9]+(e|E)(\+|-)?[0-9]+))f?|0f|[1-9][0-9]*f/`
 </div>
 <div class='syntax' noexport='true'>
-  <dfn for=syntax>uint_literal</dfn> :
+  <dfn for=syntax>hex_float_literal</dfn> :
 
-    | `/0x[0-9a-fA-F]+u|0u|[1-9][0-9]*u/`
+    | `/0x((([0-9a-fA-F]*\.[0-9a-fA-F]+|[0-9a-fA-F]+\.[0-9a-fA-F]*)((p|P)(\+|-)?[0-9]+)?)|([0-9a-fA-F]+(p|P)(\+|-)?[0-9]+))/`
 </div>
 
-Note: literals are parsed greedily. This means that for statements like `a -5`
-      this will *not* parse as `a` `minus` `5` but instead as `a` `-5` which
-      may be unexpected. A space must be inserted after the `-` if the first
-      expression is desired.
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>float_literal</dfn> :
 
-TODO(dneto): Describe how numeric literal tokens map to idealized values, and then to typed values.
+    | [=syntax/decimal_float_literal=]
+
+    | [=syntax/hex_float_literal=]
+</div>
 
 <div class='syntax' noexport='true'>
   <dfn for=syntax>const_literal</dfn> :
 
     | [=syntax/int_literal=]
 
-    | [=syntax/uint_literal=]
-
     | [=syntax/float_literal=]
 
     | [=syntax/true=]
 
     | [=syntax/false=]
 </div>
 
-<div class='syntax' noexport='true'>
-  <dfn for=syntax>float_literal</dfn> :
+When a [=numeric literal=] has a suffix, the literal denotes a value in a specific [=scalar=] type.
+Otherwise, the literal denotes a value in an idealized infinite-precision type.
 
-    | [=syntax/decimal_float_literal=]
+<table class=data>
+  <caption>Mapping literals to types</caption>
+  <thead>
+    <tr><th>Literal<th>Suffix<th>Type<th>Examples
+  </thead>
 
-    | [=syntax/hex_float_literal=]
-</div>
+  <tr><td>[=syntax/int_literal=]<td>`i`<td>[=i32=]<td>42i
+  <tr><td>[=syntax/int_literal=]<td>`u`<td>[=u32=]<td>42u
+  <tr><td>[=syntax/int_literal=]<td><td>[=IdealInteger=]<td>124
+  <tr><td>[=syntax/float_literal=]<td>`f`<td>[=f32=]<td>42f 1e5f 1.2f 0x1.0p10f
+  <tr><td>[=syntax/float_literal=]<td><td>[=IdealReal=]<td>1e5 1.2 0x1.0p10
+</table>
+
+A [=shader-creation error=] results if:
+* An [=syntax/int_literal=] with a suffix cannot be exactly represented by the target type.
+* A [=syntax/hex_float_literal=] with a `f` suffix cannot be exactly represented by the target type.
+* A [=syntax/decimal_float_literal=] with a `f` suffix overflows the target type.
+
+Note: 1.1 is not exactly representable by the [=f32=] type.
```

Comment:
Was this meant to be 1.2? I see 1.2 in examples, and both of these numbers are not exactly representable, so 1.1 appears a bit sudden.

Reply:
Or the spec could use something like "1.5" or "6.375" and avoid the weird note altogether.

Reply:
The whole point was to say that some floats that are quite concise are not exactly representable.  1.2 would do just fine. Maybe it should say *more*, that people use hex floats to be exactly sure that they get what they want, if they care.